src/main/scala/com/focaldata/cinprojectsurvey/repositories/surveys/Surveys.scala

```diff
@@ -20,34 +25,81 @@ trait Surveys {
+    def generateNickname(surveyName: Name, surveyId: SurveyId): ZIO[Any, Nothing, Nickname]
   }
 
   // The live module
   trait Live extends Surveys {
     import Queries._
 
-    // The live module service implementation
+    // The live module service implementation!
     val surveys = new Service[Env] {
-      def updateSurveyState(survey: Survey): ZIO[Env, SurveyNotFound, Survey] = for {
-        env <- ZIO.environment[Doobie]
-        updatedSurvey <- updateState(survey).option.transact(env.xa).orDie.flatMap {
-          case Some(a) => ZIO.succeed(a)
-          case None    => ZIO.fail(SurveyNotFound(survey.id))
-        }
-      } yield updatedSurvey
+      def get(surveyIdInput: SurveyIdInput): ZIO[Env, SurveyNotFound, Survey] =
+        for {
+          env <- ZIO.environment[Env]
+          survey <- querySurvey(surveyIdInput).option.transact(env.xa).orDie.flatMap {
+                     case Some(a) => ZIO.succeed(a)
+                     case None    => ZIO.fail(SurveyNotFound(surveyIdInput))
+                   }
+        } yield survey
+      def create(survey: Survey): ZIO[Env, Nothing, Survey] =
+        for {
+          env       <- ZIO.environment[Env]
+          newSurvey <- createSurvey(survey).unique.transact(env.xa).orDie
+        } yield newSurvey
+      def delete(surveyIdInput: SurveyIdInput): ZIO[Env, Nothing, Survey] =
+        for {
+          env           <- ZIO.environment[Env]
+          deletedSurvey <- deleteSurvey(surveyIdInput).unique.transact(env.xa).orDie
+        } yield deletedSurvey
+      def updateState(surveyUpdateStatusInput: SurveyUpdateStateInput, sinceDate: Instant): ZIO[Env, SurveyNotFound, Survey] =
+        for {
+          env <- ZIO.environment[Doobie]
+          updatedSurvey <- updateSurveyState(surveyUpdateStatusInput, sinceDate).option.transact(env.xa).orDie.flatMap {
+                            case Some(a) => ZIO.succeed(a)
+                            case None    => ZIO.fail(SurveyNotFound(surveyUpdateStatusInput.surveyId))
+                          }
+        } yield updatedSurvey
       def generateId: ZIO[Any, Nothing, SurveyId] = ZIO.succeed {
         // TODO should this be ZIO.effect instead of ZIO.succed? Test that.
         val uuid = UUID.randomUUID()
         new SurveyId(uuid)
       }
+      def now: ZIO[Any, Nothing, Instant] = ZIO.succeed { Instant.now() }
+
+      def generateNickname(surveyName: Name, surveyId: SurveyId): ZIO[Any, Nothing, Nickname] = ZIO.succeed { Nickname(surveyName.name + surveyId.surveyId.toString) }
     }
   }
 
   private[repositories] object Queries {
-    def updateState(survey: Survey) = sql"""update survey set sstate=${survey.state}::sstate where id=${survey.id} returning *;""".query[Survey]
+    def querySurvey(surveyIdInput: SurveyIdInput) = {
+      sql"""
+      select survey_id,
+      nickname,
+      name,
+      state,
+      created_date,
+      since_date
+      from survey where survey_id=${surveyIdInput};""".query[Survey]
```

Comment:
I think you don't need the curly brackets when you directly use the argument
```suggestion
      from survey where survey_id=$surveyIdInput""".query[Survey]
```

Reply:
changed